server/src/main/java/org/elasticsearch/index/mapper/BlockLoader.java

```diff
@@ -498,6 +509,14 @@ interface IntBuilder extends Builder {
         IntBuilder appendInt(int value);
     }
 
+    /**
+     * Specialized builder for collecting dense arrays of long values.
+     */
+    interface SingletonBulkLongBuilder extends Builder {
```

Comment:
The plan is to reuse this builder interface for other number field types too and even ordinal based fields. Given that at the codec level everything is stored as long[]. For other non long field types we need a conversion step, but that can happen in the `build()` method. For example converting to int[] by using `Math.exactInt(...)`, which can be done a simple loop in `build()` method. So I don't expect us to introduce more interfaces here.

Reply:
I think with generalizing the builder that makes sense.